### DISCUSSION TITLE
spotiifyd autdio output always headphones, spotifyd.conf overwritten at system startup

### DISCUSSION THREAD

**Discussion by daoane** [rern/rAudio#1118] [General]
spotifyd never uses the audio interface that is specified in the settings (this has been the case as long as I use rAudio/runeAudio). Thus, I specify the audio device in /etc/spotifyd.conf, and until two or three updates before the present Software Update 20221123, this worked. Since then, spotifyd.conf is always overwritten at system startup and set to the internal headphones. I encountered this problem before, but can't recall or find  a solution in the old (BBB)/new discussions. Possibly, it was solved by an update.

Has anyone encountered a similar problem and found a solution? Which would be to either
-> make spotifyd use the sound card that is specified in the rAudio settings, or
-> prevent spotifyd.conf from being overwritten at startup (why and by what is it overwritten?)

Thanks for any  help!

**Comment by jarnose** [comment]
Same problem here! Took good amount of time to find out why Spotify doens't play on my DAC. After a restart noticed that the spotifyd.conf is set back to default.

**Comment by jarnose** [comment]
Looks like the spotifyd.conf is written from /srv/http/bash/player-conf.sh with command 
`aplay -l 2> /dev/null | awk '/^card '$1'/ {print $3;exit}'`

I "fixed" this for me by editing that file so that it always uses the DAC.

**Reply by daoane** [reply]
Great, you found the file that overwrites spotifyd.conf! The path is /srv/http/bash/settings/player-conf.se (note the /settings/).
I just commented out lines 181 and 182, which overwrite spotifyd.conf and restart the service. Everything is fine now and my edited spotifyd.conf preserved.

**Reply by Patchaka** [reply]
Facing the same issue. Could you please share a screenshot of the solution

**Reply by daoane** [reply]
A screenshot won't help.

* open terminal (command shell in case you are on Windows)
* connect to your server with ssh: ssh root@<IP of your server> ; password "ros" (see front page of this repository)
* cd /srv/http/bash/settings
* nano player-conf.sh
* move to the end of the document
* comment out the second and third line from the bottom, that it looks like
`````
#-------
        fi
#       echo "$conf" > /etc/spotifyd.conf
#       systemctl try-restart spotifyd
fi
`````
* save by pressing control and x and confirm

Now /etc/spotifyd.conf won't be overwritten at startup. But you still have to edit spotifyd.conf one last time in order to make it work with your DAC. In my case, I have to replace "Headphones" with "Audio", but your device could be named differently. Then restart the spotifyd.service (systemctl restart spotifyd). Keep in mind that any future rAudio update could overwrite your edited player-conf.sh. If this happens, you have to perform the procedure again. Keep a backup of your spotifyd.conf in the same directory (spotifyd.conf.bak), then it is easy to restore. (We probably should look closer at the reason why spotifyd.conf is overwritten wrongly when there is time ...)

**Comment by Patchaka** [comment]
Huge thanks sir. Got it working now.

Regards
